Airwallex/CoreTests/AWXAPIClientTest.m

```diff
@@ -0,0 +1,79 @@
+#import "AWXConstants.h"
+#import "AWXPaymentMethodRequest.h"
+#import "AWXPaymentMethodResponse.h"
+#import "AWXTestUtils.h"
+#import "XCTestCase+Utils.h"
+#import <XCTest/XCTest.h>
+#import <OCMock/OCMock.h>
+
+@interface AWXAPIClientTest : XCTestCase
+
+@end
+
+@implementation AWXAPIClientTest
+
+- (void)testSendWithSuccess {
+    AWXGetPaymentMethodTypesRequest *request = [AWXGetPaymentMethodTypesRequest new];
+    request.active = YES;
+    request.pageNum = 0;
+    request.transactionCurrency = @"HKD";
+
+    XCTAssertEqualObjects(request.flow, @"inapp");
+
+    id sessionMock = OCMClassMock([NSURLSession class]);
+    OCMStub([sessionMock sharedSession]).andReturn(sessionMock);
+    NSData *mockData = [@"MockData" dataUsingEncoding:NSUTF8StringEncoding];
+    id responseMock = OCMClassMock([NSHTTPURLResponse class]);
+    OCMStub([responseMock statusCode]).andReturn(200);
+    OCMStub([sessionMock dataTaskWithRequest:[OCMArg isKindOfClass:[NSMutableURLRequest class]] completionHandler:([OCMArg invokeBlockWithArgs:mockData, responseMock, [NSNull null], nil])]);
+    
+    
+    AWXAPIClient *client = [[AWXAPIClient alloc] initWithConfiguration:[AWXAPIClientConfiguration sharedConfiguration]];
+    XCTestExpectation *expectation = [self expectationWithDescription:@"Get payment method list"];
+    [client send:request
+         handler:^(AWXResponse *_Nullable response, NSError *_Nullable error) {
+             XCTAssertNil(error);
+             [expectation fulfill];
+         }];
+    [self waitForExpectationsWithTimeout:60 handler:nil];
+}
+
+- (void)testSendWithFailure {
+    AWXGetPaymentMethodTypesRequest *request = [AWXGetPaymentMethodTypesRequest new];
+    request.active = YES;
+    request.pageNum = 0;
+    request.transactionCurrency = @"HKD";
+
+    XCTAssertEqualObjects(request.flow, @"inapp");
+
+    id sessionMock = OCMClassMock([NSURLSession class]);
+    OCMStub([sessionMock sharedSession]).andReturn(sessionMock);
+    NSData *mockData = [@"MockData" dataUsingEncoding:NSUTF8StringEncoding];
+    id responseMock = OCMClassMock([NSHTTPURLResponse class]);
+    OCMStub([responseMock statusCode]).andReturn(200);
+    NSError *mockError = [NSError errorWithDomain:AWXSDKErrorDomain
+                                             code:-1
+                                         userInfo:@{NSLocalizedDescriptionKey: NSLocalizedString(@"An error.", nil)}];
+    OCMStub([sessionMock dataTaskWithRequest:[OCMArg isKindOfClass:[NSMutableURLRequest class]] completionHandler:([OCMArg invokeBlockWithArgs:mockData, responseMock, mockError, nil])]);
+    
+    
+    AWXAPIClient *client = [[AWXAPIClient alloc] initWithConfiguration:[AWXAPIClientConfiguration sharedConfiguration]];
+    XCTestExpectation *expectation = [self expectationWithDescription:@"Get payment method list"];
+    [client send:request
+         handler:^(AWXResponse *_Nullable response, NSError *_Nullable error) {
+             XCTAssertNotNil(error);
```

Comment:
Same here, can check against `mockError`, so that we make sure error is passed by urlsession to this APIclient response error.